core/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -132,44 +133,53 @@ object VacuumCommand extends VacuumCommandImpl with Serializable {
       val relativizeIgnoreError =
         spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_VACUUM_RELATIVIZE_IGNORE_ERROR)
 
-      val validFiles = snapshot.stateDS
+      val allTrackedFiles = snapshot.stateDS
         .mapPartitions { actions =>
           val reservoirBase = new Path(basePath)
           val fs = reservoirBase.getFileSystem(hadoopConf.value.value)
-          actions.flatMap {
-            _.unwrap match {
+          actions.flatMap { action =>
+            val files = action.unwrap match {
               case tombstone: RemoveFile if tombstone.delTimestamp < deleteBeforeTimestamp =>
-                Nil
+                getFilePath(tombstone, fs, reservoirBase, relativizeIgnoreError)
+                  .map(f => TrackedFile(null, f))
```

Comment:
I don't think this is safe. `getFilePath` enumerates the parent paths of each file, e.g. 
```
p1=foo/p2=bar/my_file
```
would return 
```
p1=foo
p1=foo/p2=bar
p1=foo/p2=bar/my_file
```
For adds this is safe -- a subdirectory referenced by even one kept file must still be non-empty.
For deletes this is _SUPER UNSAFE_ -- we cannot prove a subdirectory is empty, by observing that one removed file no longer references it.

I'm pretty sure minor GC deletes need to just take the file name as-is, and not attempt to enumerate parent paths.

Reply:
You're right, `getFilePath` enumerates the parent paths for each file, but only empty paths will be deleted (tryDeleteNonRecursive don't delete non-empty dirs) so this should be safe. I wanted the minor GC to be aggressive and remove the empty dirs too to optimize the major GC (If we don't remove empty paths during minor GC, they will be listed and deleted during the major GC )

Reply:
Two objections...
(a) Relying on a "delete this" command to not actually delete something is scary IMO.
(b) In the common case where data grow and files get deleted because of OPTIMIZE etc, the aggressive approach will be issuing gazillions of unnecessary directory delete attempts. Aggravated by the duplication of requests pointed out elsewhere.

Reply:
makes sense, I will adjust it to delete only files found in the state.